hedera-node/src/main/java/com/hedera/services/ledger/TransactionalLedger.java

```diff
@@ -508,6 +520,71 @@ private boolean isZombie(K id) {
 		return prop -> prop.getter().apply(defaultEntity);
 	}
 
+	private List<AccountChanges<K, A, P>> getCurrentAccountChanges() {
+		final List<AccountChanges<K, A, P>> accountChanges = new ArrayList<>();
+		for (final var changesEntry : changes.entrySet()) {
+			final var accountId = changesEntry.getKey();
+			final var accountProperties = changesEntry.getValue();
+			final var changesForSingleAccount = new AccountChanges<>(accountId, toGetterTarget(accountId),
+					accountProperties);
+			accountChanges.add(changesForSingleAccount);
+		}
+
+		return accountChanges;
+	}
+
+	private void checkHbarChanges(final List<AccountChanges<K, A, P>> accountChanges) {
```

Comment:
Oops, this won't work! The record for this transaction **doesn't exist yet**...so we certainly can't look it up from a `MerkleAccount.` 😅